### PR TITLE
feat(login): add code login limit

### PR DESCRIPTION
**Pull request by longit644**
We have a limit when signing-in with `username` and `password` but with `code` is not yet, which may be the script burst code.

### CODE REVIEW

**Comment by hsluoyz** [comment]
@qianxi0410 @leo220yuyaodog @imp2002 plz review

File: object/verification.go [hunk 1]
```diff
@@ -26,6 +26,10 @@ import (
 	"xorm.io/core"
 )
 
+const (
+	wrongCode = "wrongCode"
```
Comment (leo220yuyaodog) [inline object/verification.go:30]: Herer is `wrongCode `, it is diferent from the result that function `CheckVerificationCode(dest, code, lang)` return in line 199

File: object/verification.go [hunk 2]
```diff
@@ -186,6 +190,24 @@ func DisableVerificationCode(dest string) {
 	}
 }
 
+func CheckSigninCode(user *User, dest, code, lang string) string {
+	// check the login error times
+	if msg := checkSigninErrorTimes(user, lang); msg != "" {
+		return msg
+	}
+
+	result := CheckVerificationCode(dest, code, lang)
```
Comment (leo220yuyaodog) [inline object/verification.go:199]: The result is `wrong code!`, can not go to the case `wrongCode`

**Review by leo220yuyaodog — COMMENTED** [review]
Thanks for your contribution! Please check the code, there are some problem in the pr.

File: object/verification.go (hunk 2 above)
Comment (longit644) [reply object/verification.go:199]: Thanks @leo220yuyaodog, I fixed it

**Comment by hsluoyz** [comment]
@longit644 the code is a very long and random string, can it be bursted? Can you give some example about other auth providers do such restriction?

**Comment by longit644** [comment]
Hi @hsluoyz, we only use 6 chars random string with `0-9` charset, so it can be bursted. I not tried with other providers yet, but I got a limit times when I using some application, such as: Slack.
Slack

**Pull request merged**